### ISSUE TITLE
Halyard: hal deploy connect in docker binds to localhost

### ISSUE THREAD

**Issue by spinnaker-release** [spinnaker/spinnaker#2409]
_Migrated from https://github.com/spinnaker/halyard/issues/699_

__Opened by__: @neilhwatson _(2017-09-18 20:23:53)_  Using the docker Halyard image ```hal deploy connect``` will forward remote ports to localhost in the container. These ports are not reachable from the container's host.

__Comments__:

---

@lwander _(2017-09-18 19:37:30)_: Blocked on https://github.com/kubernetes/kubernetes/issues/43962

---

@neilhwatson _(2017-09-18 19:41:38)_: I don't think this is k8s related. I'm speaking of the shipped docker image for the halyard client. Another option would be to support a native client on multiple Linux distros (e.g. Debian and RH).

---

@lwander _(2017-09-18 20:23:53)_: The connect command for k8s relies on kubectl, which is what that bug depends on.

**Comment by Thilagakumar** [comment]
I am also facing same issue. Using gke1.9.7 and spinnaker1.8.5 (tried 1.7.7 with same issue).

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by wjoel** [comment]
https://github.com/kubernetes/kubernetes/pull/46517 which fixes https://github.com/kubernetes/kubernetes/issues/43962 has been merged, so we might be able to finally do this after the next release of Kubernetes.

**Comment by spinnakerbot** [comment]
This issue is tagged as 'stale' and hasn't been updated  in 45 days, so we are tagging it as 'to-be-closed'. It will be closed in 45 days unless updates are made. If you want to remove this label, comment:

> @spinnakerbot remove-label to-be-closed

**Comment by spinnakerbot** [comment]
This issue is tagged as 'to-be-closed' and hasn't been updated  in 45 days, so we are closing it. You can always reopen this issue if needed.

**Comment by david-in-perth** [comment]
Any updates on this issue? I seem to have just encountered the same problem.

`hal --version`: 1.14.0-20190117020510
----
In first console: `docker run -p 8084:8084 -p 9000:9000     --name halyard --rm     -v ~/.hal:/home/spinnaker/.hal     -it     gcr.io/spinnaker-marketplace/halyard:stable`

In second console: `docker exec -it halyard bash` followed by `hal deploy connect`

Output:
```
+ Get current deployment
  Success
+ Connect to Spinnaker deployment.
  Success
Forwarding from 127.0.0.1:8084 -> 8084
Forwarding from 127.0.0.1:9000 -> 9000
```

In third console, `curl -sSf http://localhost:9000` gives:
> curl: (56) Recv failure: Connection reset by peer
...whereas in `docker exec -it halyard bash`, `curl -sSf http://localhost:9000` returns HTML for a Spinnaker page.

**Comment by jemc** [comment]
Same issue. I don't understand how the official docker container is usable for anyone with this problem in place.

Can this ticket be reopened? Or should I open a new ticket for the same issue?

**Comment by laszlocph** [comment]
Same here

**Comment by Lykathia** [comment]
Also just ran into this. My solution was to install `socat` and forward from the loopback interface onto the eth0 interface.

**Comment by jaekunchoi** [comment]
@Lykathia I'm having same issue. Are you able to share your `socat` command?
